platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py

```diff
@@ -40,103 +58,106 @@
-        for attr, val in platform_eeprom.__dict__.items():
-            if val is None:
-                continue
-
-            elem = _platform_eeprom_map.get(attr)
-            if elem is None:
-                continue
-
-            if isinstance(val, str):
-                value = val.replace('\0', '')
-            else:
-                value = str(val)
-
-            if attr == "sys_mfg_date":
-                value = datetime.datetime.strptime(value, '%m-%d-%y').strftime('%m/%d/%Y 00:00:00')
-
-            product = _product_dict.get(value)
-            if product is not None:
-                value = product
-            if len(eeprom_params) > 0:
-                eeprom_params += ","
-            eeprom_params += "{0:s}={1:s}".format(elem[1], value)
-
-        orig_stdout = sys.stdout
-        sys.stdout = StringIO()
+        super(Eeprom, self).__init__(_EEPROM_SYMLINK, 0, _EEPROM_STATUS, True)
+
+        self.report_status("initializing..")
+        self._eeprom_bin = bytearray()
+        if TLV_TLV_EEPROM_API_SUPPORT:
+            self._eeprom_bin = get_tlv_eeprom()
+        else:
+            eeprom_params = platfrom_eeprom_to_params(get_platform_eeprom())
+            stdout_stream = sys.stdout
+            sys.stdout = open(os.devnull, 'w')
+            self._eeprom_bin = self.set_eeprom(self._eeprom_bin, [eeprom_params])
+            sys.stdout.close()
+            sys.stdout = stdout_stream
         try:
-            eeprom_data = eeprom_tlvinfo.TlvInfoDecoder.set_eeprom(self, "", [eeprom_params])
-        finally:
-            decode_output = sys.stdout.getvalue()
-            sys.stdout = orig_stdout
-
-        eeprom_base.EepromDecoder.write_eeprom(self, eeprom_data)
-        self.__eeprom_tlv_dict = self.__parse_output(decode_output)
-
-    def __parse_output(self, decode_output):
-        EEPROM_DECODE_HEADLINES = 6
-        lines = decode_output.replace('\0', '').split('\n')
-        lines = lines[EEPROM_DECODE_HEADLINES:]
-        res = dict()
+            self.write_eeprom(self._eeprom_bin)
+            self.report_status("ok")
+        except IOError as e:
+            logging.error("Failed to write eeprom : %s" % (str(e)))
 
-        for line in lines:
-            try:
-                # match whitespace-separated tag hex, length and value (value is mathced with its whitespaces)
-                match = re.search('(0x[0-9a-fA-F]{2})([\s]+[\S]+[\s]+)([\S]+[\s]*[\S]*)', line)
-                if match is not None:
-                    code = match.group(1)
-                    value = match.group(3).rstrip('\0')
-                    res[code] = value
-            except Exception:
-                pass
-        return res
+        visitor = SavingTlvVisitor()
+        self.visit_eeprom(self._eeprom_bin, visitor)
+        self._system_eeprom_info = visitor.get_contents()
 
-    def __tlv_get(self, code):
-        return self.__eeprom_tlv_dict.get("0x{:X}".format(code), 'N/A')
+    def get_contents(self):
```

Comment:
Probably we can split into two methods like
```
get_raw_data()
```
```
get_data()
```

Reply:
Done